### PR TITLE
[docs] add link to ML Flow and treesnip in LightGBM docs

### PR DESCRIPTION
**Pull request by jameslamb**
This PR adds a link to the LightGBM integration with [ML Flow](https://mlflow.org/docs/latest/index.html). If you're curious, you can see more details on the individual methods at https://mlflow.org/docs/latest/python_api/mlflow.lightgbm.html#module-mlflow.lightgbm.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -94,6 +94,8 @@ Ruby gem: https://github.com/ankane/lightgbm
 
 LightGBM4j (Java high-level binding): https://github.com/metarank/lightgbm4j
 
+ML Flow (experiment tracking, model monitoring framework): https://mlflow.org/docs/latest/models.html#lightgbm-lightgbm
```
Comment (StrikerRUS) [inline README.md:97]: The format of this list expects GitHub link.
```suggestion
ML Flow (experiment tracking, model monitoring framework): https://github.com/mlflow/mlflow
```

**Comment by StrikerRUS** [comment]
@jameslamb As we are touching this list now, could you please add some R-related projects (I'm not familiar with them at all)?
Maybe some from `tydimodels` ecosystem ? 
https://github.com/curso-r/treesnip
https://github.com/tidymodels/tidymodels

**Comment by jameslamb** [comment]
> @jameslamb As we are touching this list now, could you please add some R-related projects (I'm not familiar with them at all)?
> Maybe some from `tidymodels` ecosystem ?
> https://github.com/curso-r/treesnip
> https://github.com/tidymodels/tidymodels

I don't think that `{tidymodels}` has any LightGBM integration. Searching "lightgbm" in their repo returns 0 results.

I'll add `{treesnip}` though

**Review by StrikerRUS — APPROVED** [review]
Thanks!

**Comment by t-wojciech** [comment]
There is also `lightgbm` for [mlr3](https://mlr3.mlr-org.com/) framework: https://github.com/mlr3learners/mlr3learners.lightgbm

**Comment by jameslamb** [comment]
Thanks @t-wojciech . Wasn't familiar with `{mlr3}`. I'd welcome a pull request to add that link

**Pull request merged**